### DISCUSSION TITLE
Getting error while running our end user journey scenario in artillery :- worker error, id: 1 TypeError [Error]: k.toLowerCase is not a function

### DISCUSSION THREAD

**Discussion by nishant-shah-social** [artilleryio/artillery#1570] [Artillery]
I have written down an artillery scenario to emulate end user journey of our product..
The scenario comprises of around 20-25 requests calls + accessing data from external csv file + at certain places I had to write custom JS functions and call them in the yaml file via beforeRequest tag..
Coming to the Point :- when I try to run the scenario, I am getting an error 
```
worker error, id: 1 TypeError [Error]: k.toLowerCase is not a function
    at /usr/local/lib/node_modules/artillery/core/lib/engine_http.js:853:14
    at /usr/local/lib/node_modules/artillery/node_modules/lodash/lodash.js:13874:16
    at arrayEach (/usr/local/lib/node_modules/artillery/node_modules/lodash/lodash.js:530:11)
    at Function.transform (/usr/local/lib/node_modules/artillery/node_modules/lodash/lodash.js:13873:43)
    at lowcaseKeys (/usr/local/lib/node_modules/artillery/core/lib/engine_http.js:852:12)
    at done (/usr/local/lib/node_modules/artillery/core/lib/engine_http.js:389:11)
    at /usr/local/lib/node_modules/artillery/node_modules/async/dist/async.js:473:16
    at replenish (/usr/local/lib/node_modules/artillery/node_modules/async/dist/async.js:1009:25)
    at /usr/local/lib/node_modules/artillery/node_modules/async/dist/async.js:1019:9
    at eachLimit$1 (/usr/local/lib/node_modules/artillery/node_modules/async/dist/async.js:3199:24)
```
Nothing in this whole stack trace gives me even a slightest of the hint at what line the error might be present, either in yaml file or the custom js file...
I tried using `Debug=http` flag, which printed the Request Body of first 3 requests and then just hung after printing the above same error. So, I am suspecting the error might be in the 4th request...However, its really difficult to understand/troubleshoot from this error message as it completely mentions about artillery source code files..
Is there any better way of debugging?

Thanks,
Nishant Shah

**Comment by hassy** [comment]
hi @nishant-shah-social 👋 

`DEBUG=http` is meant for debugging HTTP-level issues, e.g. the service returning data in a format that you don't expect. The error you're running into is due to an exception inside Artillery itself.

Does your 4th request set a custom header? Having looked at the code in `engine_http.js` around line 853 (mentioned in the stack trace you shared), the issue is most likely with something to do with that.

**Comment by nishant-shah-social** [comment]
Thanks @hassy for the quick reply.
Yes, I understand DEBUG option is for debugging http-level issues. But with the error message I got it was really difficult for me to understand in which out of the 20-25 request calls the issue might be. So, I used it to identify at what point the execution stops and so that I can narrow down on that particular request :)
However, thanks for pointing me to the Headers...I checked and I had specified headers in the 4th request as an array whereas its expected to be an object. I will execute after making the changes later today and hopefully it should solve this problem..

However, just curious to know...Is it somehow possible to know easily that the way I specified headers was incorrect? Can this be somehow handled in Artillery source code and a more relevant error message be returned so that it becomes easy to debug...
Mainly when automating end user journeys one has to do lot of request chaining and response extraction and one will definitely end up writing some custom js as well....So, its likely that one will make silly or non-silly mistakes and if the tool pinpoints the error correctly, it  will surely help us move faster.

Thanks,
Nishant Shah
